spec/migrations/use_encrypted_data_for_account_ose_secret_spec.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+migration_dir = 'db/migrate/'
+migration_file_name = '20220107115502_use_encrypted_data_for_account_ose_secret.rb'
+mig_file = Dir[Rails.root.join(migration_dir + migration_file_name)].first
+require mig_file
+
+describe UseEncryptedDataForAccountOseSecret do
+
+  let(:migration) { described_class.new }
+  let(:folder1) { folders(:folder1) }
+
+  def silent
+    verbose = ActiveRecord::Migration.verbose = false
+
+    yield
+
+    ActiveRecord::Migration.verbose = verbose
+  end
+
+  around do |test|
+    silent { test.run }
+  end
+
+  context 'up' do
+    before do
+      migration.down
+      @ose_secret_account = LegacyOSESecret.create!(name: 'secret1',
+                                                    folder_id: folder1.id,
+                                                    type: 'Account::OSESecret',
+                                                    ose_secret: decoded_secret)
+
+      @ose_secret_account.encrypt(team1_password)
+      @ose_secret_account.save
+    end
+
+    it 'migrates ose_secret credentials to EncryptedData with base64 encoding' do
+      migration.up
+
+      ose_secret_account = Account.find(@ose_secret_account.id)
+      ose_secret_account.decrypt(team1_password)
+
+      expect(ose_secret_account.ose_secret).to eq(decoded_secret)
+    end
+  end
+
+  context 'down' do
+    after { migration.up }
+
+    it 'reverts to previous schema' do
+      ose_secret_account = Account::OSESecret.create!(name: 'secret1',
+                                                      folder_id: folder1.id,
+                                                      ose_secret: decoded_secret)
+
+      ose_secret_account.encrypt(team1_password)
+      ose_secret_account.save
+
+      pre_mig_iv = Base64.strict_encode64(ose_secret_account.encrypted_data[:ose_secret][:iv])
+
```

Comment:
shouldn't we also test the migration back of an ose secret with legacy datastructure ?